CONTRIBUTING.MD

```diff
@@ -0,0 +1,27 @@
+# Contributing a Devfile Stack to this registry
+
+This document outlines the requirements for contributing a devfile stack to this repository
+
+## Steps
+
+1) Verify your Devfile stack functions with odo.
+  
+    - Core odo functions such as `odo create --devfile <devfile.yaml>`, `odo push`, `odo url create` should work with the devfile.
+    - PR tests on this repository will verify this functionality as well.
+
+2) Add a folder for the stack to `stacks/` in this repository.
+  
+    - Name format should be in the form of `<language>-<framework>`. E.g. `java-quarkus`, `python-django`, etc.
+
+3) Add the devfile.yaml and any other necessary files for the stack under the stack folder.
+
+4) Run the `.ci/build.sh` to build the registry into a container image.
+  
+    - This will also validate the devfiles in this repository, making sure they conform to a minimum standard.
+    - This step will also be run in the PR build for the repository.
+
+5) (Optional) Push the container image to a container registry and [deploy](https://github.com/devfile/registry-support#deploy) on to Kubernetes or OpenShift.
```

Comment:
What is this step for? Is this step for #1 to test out the stack? If it is, should it go before step 1?

Reply:
It's a separate step. Since odo lets you test stacks/devfiles without a registry (e.g. with the `--devfile` flag), the stack can be tested without building it into a container image.

This step is just an added level of testing if stack authors want to verify their stack as part of an OCI registry as well. But this'll also be tested in the CI build (once https://github.com/openshift/odo/pull/4525 is merged), so it's not strictly required.